sdk/servicebus/service-bus/src/receivers/shared.ts

```diff
@@ -218,49 +233,14 @@ function settleMessage(
     );
     throw error;
   }
-  const isDeferredMessage = !message.delivery.link;
-  const receiver = isDeferredMessage
-    ? undefined
-    : context.getReceiverFromCache(message.delivery.link.name, message.sessionId);
-  const associatedLinkName = receiver?.name;
-
-  if (!isDeferredMessage) {
-    // In case the message wasn't from a deferred queue,
-    //   1. We can verify the remote_settled flag on the delivery
-    //      - If the flag is true, throw an error since the message has been settled (Specifically, with a receive link)
-    //      - If the flag is false, we can't say that the message has not been settled
-    //        since settling with the management link won't update the delivery (In this case, service would throw an error)
-    //   2. If the message has a session-id and if the associated receiver link is unavailable,
-    //      then throw an error since we need a lock on the session to settle the message.
-    let error: Error | undefined;
-    if (message.delivery.remote_settled) {
-      error = new Error(`Failed to ${operation} the message as this message is already settled.`);
-    } else if ((!receiver || !receiver.isOpen()) && message.sessionId != undefined) {
-      error = translateServiceBusError({
-        description:
-          `Failed to ${operation} the message as the AMQP link with which the message was ` +
-          `received is no longer alive.`,
-        condition: ErrorNameConditionMapper.SessionLockLostError
-      });
-    }
-    if (error) {
-      receiverLogger.logError(
-        error,
-        "[%s] An error occurred when settling a message with id '%s'",
-        context.connectionId,
-        message.messageId
-      );
-      throw error;
-    }
-  }
 
   // Message Settlement with managementLink
   // 1. If the received message is deferred as such messages can only be settled using managementLink
   // 2. If the associated receiver link is not available. This does not apply to messages from sessions as we need a lock on the session to do so.
   if (isDeferredMessage || ((!receiver || !receiver.isOpen()) && message.sessionId == undefined)) {
     return context
       .getManagementClient(entityPath)
-      .updateDispositionStatus(message.lockToken, operation, {
+      .updateDispositionStatus(message.lockToken!, operation, {
```

Comment:
Are we guaranteed that `lockToken` is defined because we're working with a deferred message?

Reply:
We expect `throwErrorIfInvalidOperationOnMessage()` to be called before trying to settle the message which will check for the lock token